seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java

```diff
@@ -67,8 +67,13 @@ public ClientJobExecutionEnvironment(
         this.seaTunnelHazelcastClient = seaTunnelHazelcastClient;
         this.jobClient = new JobClient(seaTunnelHazelcastClient);
         this.seaTunnelConfig = seaTunnelConfig;
-        this.jobConfig.setJobContext(
-                new JobContext(isStartWithSavePoint ? jobId : jobClient.getNewJobId()));
+        Long finalJobId;
+        if (isStartWithSavePoint || jobId != null) {
+            finalJobId = jobId;
```

Comment:
@Hisoka-X hi, i relealize a problem, if user pass same job id with the running jobs, will it has issue? 
now i story some status in IMAP by job id. if use same job id, the status maybe impact?

Reply:
Yes, the key point is that the ID cannot be repeated, which currently needs to be ensured by the user

Reply:
Fix by https://github.com/apache/seatunnel/pull/7021